static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.js

```diff
@@ -122,26 +123,34 @@ export const init = (): Promise<void> => {
                     npaFlag =
                         Object.keys(state.tcfv2.consents).length === 0 ||
                         Object.values(state.tcfv2.consents).includes(false);
+                    canRun = state.tcfv2.consents['1']; // Store and/or access information on a device
                 } else {
                     // TCFv1 mode
                     npaFlag = Object.values(state).includes(false);
+                    // TODO: decide if purpose 1 should apply to TCFv1
                 }
                 window.googletag.cmd.push(() => {
                     window.googletag
                         .pubads()
                         .setRequestNonPersonalizedAds(npaFlag ? 1 : 0);
                 });
             }
-        });
 
-        // Just load googletag. Prebid will already be loaded, and googletag is already added to the window by Prebid.
-        return loadScript(
-            config.get(
-                'libs.googletag',
-                '//www.googletagservices.com/tag/js/gpt.js'
-            ),
-            { async: false }
-        );
+            // Load googletag if TCFv2 purpose 1 consent is not `false`.
+            // Prebid will already be loaded, and window.googletag is stubbed in `commercial.js`.
+            // TODO: what if Prebid is also behind consent?
+            if (canRun) {
+                return loadScript(
+                    config.get(
+                        'libs.googletag',
+                        '//www.googletagservices.com/tag/js/gpt.js'
+                    ),
+                    { async: false }
+                );
+            }
+            // return Promise.resolve();
```

Comment:
Can remove this comment 🧹 🙂

Reply:
Yes: I'm still a little shaky on where exactly we should return `Promise.resolve()`

Reply:
I have refactored it a little bit to load the script inside the callback if it's TCFv2 and move the Promise.resolve